src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/PaperCaseCreateContestedMidHandler.java

```diff
@@ -0,0 +1,46 @@
+package uk.gov.hmcts.reform.finrem.caseorchestration.handler;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.finrem.caseorchestration.ccd.callback.CallbackType;
+import uk.gov.hmcts.reform.finrem.caseorchestration.controllers.GenericAboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.finrem.caseorchestration.mapper.FinremCaseDetailsMapper;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.EventType;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CaseType;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.FinremCaseData;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.FinremCaseDetails;
+import uk.gov.hmcts.reform.finrem.caseorchestration.service.InternationalPostalService;
+
+@Slf4j
+@Service
+public class PaperCaseCreateContestedMidHandler extends FinremCallbackHandler {
+
+    private final InternationalPostalService postalService;
+
+    public PaperCaseCreateContestedMidHandler(FinremCaseDetailsMapper finremCaseDetailsMapper,
+                                              InternationalPostalService postalService) {
+        super(finremCaseDetailsMapper);
+        this.postalService = postalService;
+    }
+
+
+    @Override
+    public boolean canHandle(CallbackType callbackType, CaseType caseType, EventType eventType) {
+        return CallbackType.MID_EVENT.equals(callbackType)
+            && CaseType.CONTESTED.equals(caseType)
+            && EventType.NEW_PAPER_CASE.equals(eventType);
+    }
+
+    @Override
+    public GenericAboutToStartOrSubmitCallbackResponse<FinremCaseData> handle(FinremCallbackRequest callbackRequest,
+                                                                              String userAuthorisation) {
+        FinremCaseDetails caseDetails = callbackRequest.getCaseDetails();
+        log.info("Invoking contested event {} mid event callback", EventType.NEW_PAPER_CASE);
```

Comment:
Missing caseId in the log message

Reply:
Case hasn't created so do not have caseId